plugins/extractors/maxcompute/maxcompute.go

```diff
@@ -0,0 +1,250 @@
+	e.ScopeNotRequired = true
+
+	return e
+}
+
+func (e *Extractor) Init(ctx context.Context, config plugins.Config) error {
+	return e.BaseExtractor.Init(ctx, config)
+}
+
+func (e *Extractor) Extract(_ context.Context, emit plugins.Emit) error {
+	aliAccount := account.NewAliyunAccount(e.config.AccessKey.ID, e.config.AccessKey.Secret)
+	odpsIns := odps.NewOdps(aliAccount, e.config.EndpointProject)
+	project := odpsIns.Project(e.config.ProjectName)
+
+	schemas := project.Schemas()
+
+	err := schemas.List(func(schema *odps.Schema, err error) {
+		if err != nil {
+			e.logger.Error("failed to list schemas", "error", err)
+			return
+		}
+		if e.config.SchemaName != "" && schema.Name() != e.config.SchemaName {
+			return
+		}
+		if contains(e.config.Exclude.Schemas, schema.Name()) {
+			return
+		}
+
+		newIns := odps.NewOdps(aliAccount, e.config.EndpointProject)
+		newIns.SetCurrentSchemaName(schema.Name())
+		newIns.SetDefaultProjectName(e.config.ProjectName)
+		newProj := newIns.Project(e.config.ProjectName)
+
+		e.fetchTablesFromSchema(newProj, emit)
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (e *Extractor) fetchTablesFromSchema(project *odps.Project, emit plugins.Emit) {
+	var wg sync.WaitGroup
+	project.Tables().List(
+		func(t *odps.Table, err error) {
+			wg.Add(1)
+			if err != nil {
+				e.logger.Error("table list", err)
+			}
+			if contains(e.config.Exclude.Tables, fmt.Sprintf("%s.%s", t.SchemaName(), t.Name())) {
+				wg.Done()
+				return
+			}
+			go e.processTable(t, emit, &wg)
+		},
+	)
+
+	wg.Wait()
+}
+
+func (e *Extractor) processTable(table *odps.Table, emit plugins.Emit, wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	err := table.Load()
+	if err != nil {
+		e.logger.Error("failed to get table info", "table", table.Name(), "error", err)
+		return
+	}
+
+	fmt.Println(table.Name())
+
+	asset, err := e.buildAsset(table)
+	if err != nil {
+		e.logger.Error("failed to build asset", "table", table.Name(), "error", err)
+		return
+	}
+
+	emit(models.NewRecord(asset))
+}
+
+func (e *Extractor) buildAsset(tableInfo *odps.Table) (*v1beta2.Asset, error) {
+	defaultSchema := "default"
+	schemaName := tableInfo.SchemaName()
+	if schemaName == "" {
+		schemaName = defaultSchema
+	}
+
+	tableURN := plugins.MaxComputeURN(tableInfo.ProjectName(), schemaName, tableInfo.Name())
+
+	schema := tableInfo.Schema()
+	asset := &v1beta2.Asset{
+		Urn:         tableURN,
+		Name:        schema.TableName,
+		Type:        tableInfo.Type().String(),
+		Description: schema.Comment,
+		CreateTime:  timestamppb.New(tableInfo.CreatedTime()),
+		UpdateTime:  timestamppb.New(tableInfo.LastModifiedTime()),
+		Service:     "maxcompute",
+		// Labels: tableInfo.TableLabel(),
+	}
+
+	attributesData := buildAttributesData(tableInfo)
+
+	var columns []*v1beta2.Column
+	for _, col := range schema.Columns {
+		columnData := &v1beta2.Column{
+			Name:        col.Name,
+			DataType:    col.Type.Name(),
```

Comment:
if the type is struct, please add the subcolumn recursively as discussed

Reply:
addressed